### PR TITLE
Tape recorder revamp

### PR DESCRIPTION
**Pull request by Hopekz**
Old way 
------
(Old way still works):
0YXgVcMNdv

New way:
-------

**Context-sensitive buttons!**

WJBmNrbmGs

While recording:
dreamseeker_ymEvPOT8ZM

while playing:
dreamseeker_se9oLhHKMV

**And you can also alt-click a tape recorder to play it like in #8445**
UEl2rkUv3y

#### Changelog

:cl:  Hopek
rscadd: Tape recorder has been reworked.
/:cl:

### CODE REVIEW

**Comment by MegaEmpirical** [comment]
Nice.

**Review by JamieD1 — APPROVED** [review]

File: code/game/objects/items/devices/taperecorder.dm [hunk 1]
```diff
@@ -32,6 +34,30 @@
...
+			icons_available += list("Record" = image(icon = icon_directory, icon_state = "record"))
+	
+	if(playing)
+		icons_available += list("Pause" = image(icon = icon_directory, icon_state = "pause"))
+	else
+		if(!recording)
+			icons_available += list("Play" = image(icon = icon_directory, icon_state = "play"))
+	
+	if(canprint && !recording && !playing)
+		icons_available += list("Print Transcript" = image(icon = icon_directory, icon_state = "print"))
```
Comment (alexkar598) [inline code/game/objects/items/devices/taperecorder.dm]:
```suggestion
	if(recording)
		icons_available += list("Stop Recording" = image(icon = icon_directory, icon_state = "record_stop"))

	if(playing)
		icons_available += list("Pause" = image(icon = icon_directory, icon_state = "pause"))

	if(!playing && !recording)
		icons_available += list("Record" = image(icon = icon_directory, icon_state = "record"))
		icons_available += list("Play" = image(icon = icon_directory, icon_state = "play"))
		if(canprint)
			icons_available += list("Print Transcript" = image(icon = icon_directory, icon_state = "print"))
```

File: code/game/objects/items/devices/taperecorder.dm [hunk 2]
```diff
@@ -194,12 +220,29 @@
...
+		if(selection == "Stop Recording" || selection == "Pause")
+			stop()
+		if(selection == "Record")
+			record()
+		if(selection == "Play")
+			play()
+		if(selection == "Print Transcript")
+			print_transcript()
+		if(selection == "Eject")
+			eject(user)
```
Comment (alexkar598) [inline code/game/objects/items/devices/taperecorder.dm]:
```suggestion
		switch(selection)
			if("Stop Recording" || "Pause")
				stop()
			if("Record")
				record()
			if("Play")
				play()
			if("Print Transcript")
				print_transcript()
			if("Eject")
				eject(user)
```

**Review by alexkar598 — DISMISSED** [review]

**Comment by alexkar598** [comment]
Stale

**Comment by JamieD1** [comment]
You incorporated alexkars changes into your tg pr but didn't commit them here..

Hmmm?

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**